packages/contracts/src/utils/exchange_wrapper.ts

```diff
@@ -142,21 +144,31 @@ export class ExchangeWrapper {
     ): Promise<TransactionReceiptWithDecodedLogs> {
         const params = formatters.createMarketFillOrders(orders, opts.takerTokenFillAmount);
         const txHash = await this._exchange.marketFillOrders.sendTransactionAsync(
-            params.orderAddresses,
-            params.orderValues,
+            params.orders,
+            params.takerTokenFillAmount,
+            params.signatures,
+            { from },
+        );
+        const tx = await this._zeroEx.awaitTransactionMinedAsync(txHash);
+        tx.logs = _.filter(tx.logs, log => log.address === this._exchange.address);
+        tx.logs = _.map(tx.logs, log => this._logDecoder.decodeLogOrThrow(log));
```

Comment:
This is still repeated several times in this file. Please move out into helper function that takes a `tx` and return a `tx` that is filtered and decoded.